### PR TITLE
Refactored Search Page

### PR DESCRIPTION
**Pull request by taryarwinhtet**
Separated everything into widget and method

### CODE REVIEW

File: lib/search/search_page.dart [hunk 1]
```diff
@@ -198,105 +95,260 @@ class SearchPage extends ConsumerWidget {
...
-                          SearchHistory(selectedRef),
-                        ],
-                      ),
-                    ),
-                  ),
+                  SearchHistoryWidget(),
                 ])));
   }
+
+  Container SearchButton(BuildContext context, void setSearchValue()) {
```
Comment (kovarny) [inline lib/search/search_page.dart:111]: please follow the dart notation guidelines, and let's prefix build methods with 'build', e.g. buildSearchButton

File: lib/search/search_page.dart [hunk 2]
```diff
@@ -198,105 +95,260 @@ class SearchPage extends ConsumerWidget {
...
+                                        hintText: 'Enter Name Here',
+                                      ),
+                                    ),
+                                  ),
+                                ),
+                              );
+  }
+}
+
+class SearchPageInfo extends StatelessWidget {
```
Comment (kovarny) [inline lib/search/search_page.dart:200]: pls move widgets to separate files.

**Review by kovarny — CHANGES_REQUESTED** [review]

File: lib/search/search_page.dart (hunk 2 above)
Comment (taryarwinhtet) [reply lib/search/search_page.dart]: resolved

File: lib/search/matches_ui.dart [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:screensite/search/search_details.dart';
+import 'package:screensite/search/search_history_ui.dart';
+
+class MatchesUI extends ConsumerWidget {
```
Comment (kovarny) [inline lib/search/matches_ui.dart:7]: MatchesWidget

File: lib/search/profile_information_ui.dart [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:screensite/search/search_results_item.dart';
+import 'package:screensite/search/search_history_ui.dart';
+
+class ProfileInformationUI extends ConsumerWidget {
```
Comment (kovarny) [inline lib/search/profile_information_ui.dart:6]: Please replace UI with Widget

**Review by kovarny — CHANGES_REQUESTED** [review]

File: lib/search/search_page.dart [hunk 5]
```diff
@@ -76,119 +80,16 @@ class SearchPage extends ConsumerWidget {
...
-                                        //   'timeCreated':
-                                        //       FieldValue.serverTimestamp(),
-                                        //   'author': FirebaseAuth
-                                        //       .instance.currentUser!.uid,
-                                        // });
-
-                                        setSearchValue();
-                                      }),
-                                )
+                                buildSearchBar(setSearchValue),
```
Comment (kovarny) [inline lib/search/search_page.dart:91]: I don't think this is the right way of doing it. Why do we need to pass a local function? Instead let's have it as a method that can be called in those methods.

**Review by kovarny — CHANGES_REQUESTED** [review]

**Review by kovarny — APPROVED** [review]

**Pull request closed without merging**